### PR TITLE
Remove connect_type setting

### PR DESCRIPTION
**Pull request by nijel**
It is really not necessary as MySQL decides connection type rather based
on hostname than on anything else.

Signed-off-by: Michal Čihař <michal@cihar.com>

Before submitting pull request, please check that every commit:

- [x] Has proper Signed-Off-By
- [x] Has commit message which describes it
- [x] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [x] Any new functionality is covered by tests

### CODE REVIEW

**Comment by ibennetch** [comment]
If, for some reason, we don't merge this, we should address an interesting situation where a user must use connect_type='socket' in order for `$cfg['Servers'][$i]['socket']` to take effect. It makes sense why it was implemented this way, but if this doesn't get merged we can improve that.

**Comment by ibennetch** [comment]
Anyway, I like this and think it's a good improvement. I don't see a downside of merging this and it works for me.

**Review by ibennetch — APPROVED** [review]

File: doc/config.rst [hunk 1]
```diff
@@ -219,7 +219,9 @@ Server connection settings
     .. note::
 
         The hostname ``localhost`` is handled specially by MySQL and it
-        ignores :config:option:`$cfg['Servers'][$i]['port']` in this case.
+        ignores :config:option:`$cfg['Servers'][$i]['port']` in this case
```
Comment (ibennetch) [inline doc/config.rst]: I'd make this "The hostname ``localhost`` is handled specially by MySQL and it uses the socket based connection protocol. To use TCP/IP networking, use an IP address or hostname such as ``127.0.0.1`` or ``db.example.com``. You can configure socket address by :config:option:`$cfg['Servers'][$i]['socket']`."

File: doc/config.rst [hunk 2]
```diff
@@ -418,6 +425,13 @@ Server connection settings
     :type: string
     :default: ``'tcp'``
 
+    .. deprecated:: 4.7.0
+
+       This settings is no longer used as of 4.7.0, the MySQL anyway decides
```
Comment (ibennetch) [inline doc/config.rst]: I suggest "This setting is no longer used as of 4.7.0, since MySQL decides the"

File: doc/config.rst [hunk 3]
```diff
@@ -418,6 +425,13 @@ Server connection settings
...
+       connection type based on host, so it could lead to unexpected results.
+       Please set :config:option:`$cfg['Servers'][$i]['host']` accordingly
```
Comment (ibennetch) [inline doc/config.rst]: It's helpful how you direct users to 'host' here instead, good job.

**Review by ibennetch — CHANGES_REQUESTED** [review]
I have a few suggestions for documentation language.

**Comment by nijel** [comment]
Applied your suggestions in 34946a285b.

**Comment by ibennetch** [comment]
Great, I think this looks ready.

Do you really want to apply it to QA_4_7? I don't object because the modifications are relatively minor, but in general I think we should avoid such changes to a QA branch.

**Comment by nijel** [comment]
I'm not sure, but I thought it is still reasonable to do that for 4.7 now.

**Comment by ibennetch** [comment]
Sure, let's go ahead with putting it in 4.7.

**Comment by nijel** [comment]
Okay, merging.

**Comment by ibennetch** [comment]
Great. Did you use `--author` to add my name to this commit or is that part of the new Github review feature? Either way, it's pretty helpful!

**Comment by nijel** [comment]
I've used commandline with ``-author``

**Pull request merged**